eessi/testsuite/tests/apps/gromacs.py

```diff
@@ -35,43 +35,42 @@
 from hpctestlib.sciapps.gromacs.benchmarks import gromacs_check
 
 from eessi.testsuite import hooks
-from eessi.testsuite.constants import SCALES, TAGS
+from eessi.testsuite.constants import COMPUTE_UNIT, DEVICE_TYPES, SCALES
+from eessi.testsuite.eessi_mixin import EESSI_Mixin
 from eessi.testsuite.utils import find_modules, log
 
 
+class EESSI_GROMACS_base(gromacs_check):
+    @run_after('init')
+    def set_device_type(self):
+        self.device_type = self.nb_impl
+
+
 @rfm.simple_test
-class EESSI_GROMACS(gromacs_check):
+class EESSI_GROMACS(EESSI_GROMACS_base, EESSI_Mixin):
     scale = parameter(SCALES.keys())
-    valid_prog_environs = ['default']
-    valid_systems = ['*']
     time_limit = '30m'
     module_name = parameter(find_modules('GROMACS'))
+    bench_name_ci = 'HECBioSim/Crambin'
+    # measure_memory_usage = True
 
-    @run_after('init')
-    def run_after_init(self):
-        """Hooks to run after the init phase"""
-
-        # Filter on which scales are supported by the partitions defined in the ReFrame configuration
-        hooks.filter_supported_scales(self)
-
-        # Make sure that GPU tests run in partitions that support running on a GPU,
-        # and that CPU-only tests run in partitions that support running CPU-only.
-        # Also support setting valid_systems on the cmd line.
-        hooks.filter_valid_systems_by_device_type(self, required_device_type=self.nb_impl)
+    def required_mem_per_node(self):
+        return self.num_tasks_per_node * 1024
 
-        # Support selecting modules on the cmd line.
-        hooks.set_modules(self)
-
-        # Support selecting scales on the cmd line via tags.
-        hooks.set_tag_scale(self)
+    def __init__(self):
+        # self.device_type must be set before the @run_after('init') hooks of the EESSI_Mixin class
+        self.device_type = self.nb_impl
 
     @run_after('init')
-    def set_tag_ci(self):
-        """Set tag CI on smallest benchmark, so it can be selected on the cmd line via --tag CI"""
-        # Crambin input is smallest input (20K atoms), cfr. https://www.hecbiosim.ac.uk/access-hpc/benchmarks
-        if self.benchmark_info[0] == 'HECBioSim/Crambin':
-            self.tags.add(TAGS['CI'])
-            log(f'tags set to {self.tags}')
+    def set_compute_unit(self):
+        """Set the compute unit to which tasks will be assigned"""
+        if self.device_type == DEVICE_TYPES['CPU']:
```

Comment:
Maybe comment what we intend here: assign one task per CPU core for CPU runs, and one task per GPU for GPU runs.